test_vm/tests/change_owner_test.rs

```diff
@@ -1,163 +1,271 @@
-use fvm_shared::error::ExitCode;
 use fvm_shared::sector::RegisteredSealProof;
+use fvm_shared::state::StateTreeVersion;
+use fvm_shared::version::NetworkVersion;
+use fvm_workbench_api::wrangler::ExecutionWrangler;
+use fvm_workbench_api::WorkbenchBuilder;
+use fvm_workbench_builtin_actors::genesis::{create_genesis_actors, GenesisSpec};
+use fvm_workbench_vm::builder::FvmBenchBuilder;
+use fvm_workbench_vm::externs::FakeExterns;
+use test_vm::bench::Benchmarker;
 use test_vm::util::{
-    apply_code, change_beneficiary, change_owner_address, create_accounts, create_miner,
+    apply_code, change_beneficiary, change_beneficiary_, change_owner_address,
+    change_owner_address_, create_accounts, create_accounts_, create_miner, create_miner_,
     get_beneficiary,
 };
-use test_vm::VM;
+use test_vm::TestVM;
 
-#[test]
-fn change_owner_success() {
-    let store = MemoryBlockstore::new();
-    let mut v = VM::new_with_singletons(&store);
-    let addrs = create_accounts(&v, 3, TokenAmount::from_whole(10_000));
+enum ConcreteVM<'vm, 'bs> {
+    TestVM(&'vm TestVM<'bs>),
+    BenchmarkVM(&'vm Benchmarker<'vm>),
+}
+
+/// Sample test that can be injected with either a TestVM (rust execution, stack traceable etc) or a
+/// _BenchmarkVM (wasm execution, gas metering etc)
+fn change_owner_test(concrete_vm: ConcreteVM) {
+    let vm = match concrete_vm {
+        ConcreteVM::TestVM(tvm) => tvm.vm(),
+        ConcreteVM::BenchmarkVM(vm) => vm,
+    };
+
+    // when the tests cases are merged, remove the match and use the concrete_vm directly
+    let addrs = create_accounts_(vm, 3, TokenAmount::from_whole(10_000), SignatureType::BLS);
     let seal_proof = RegisteredSealProof::StackedDRG32GiBV1P1;
     let (owner, worker, new_owner, beneficiary) = (addrs[0], addrs[0], addrs[1], addrs[2]);
 
     // create miner
-    let miner_id = create_miner(
-        &mut v,
+    let miner_id = create_miner_(
+        vm,
         owner,
         worker,
         seal_proof.registered_window_post_proof().unwrap(),
         TokenAmount::from_whole(1_000),
     )
     .0;
 
-    change_beneficiary(
-        &v,
+    change_beneficiary_(
+        vm,
         owner,
         miner_id,
         &ChangeBeneficiaryParams::new(beneficiary, TokenAmount::from_atto(100), 100),
     );
-    change_owner_address(&v, owner, miner_id, new_owner);
-    let miner_info = v.get_miner_info(miner_id);
-    assert_eq!(new_owner, miner_info.pending_owner_address.unwrap());
+    change_owner_address_(vm, owner, miner_id, new_owner);
+    match concrete_vm {
+        ConcreteVM::TestVM(tvm) => {
+            let miner_info = tvm.get_miner_info(miner_id);
+            assert_eq!(new_owner, miner_info.pending_owner_address.unwrap());
+        }
+        #[cfg(feature = "benchmark")]
+        ConcreteVM::_BenchmarkVM(_) => {
+            // this block shouldn't compile unless the feature is enabled
+            compilation_error;
+        }
+        _ => {}
+    }
```

Comment:
Probably can write a macro to clean this up. For most cases I imagine we'd do assertions on the TestVM only. 

The BenchmarkVm can be interrogated at a higher level to analyse the gas traces.

Reply:
- I think we will want to leave the assertions in place, at least to start. If we later decide that they'll interfere too much with gas benchmarking, we can abstract them later.
- The VM trait will need a method to read state from its blockstore (probably just exposing a blockstore getter). But then helper methods like "get_miner_info" will need to become free functions rather than methods of any VM.
- We should end up with almost no evidence of alternative VMs or the possibility of the tests being used for benchmarking etc, so no `cfg` macros and no enums/matches. Just the `&dyn VM` abstraction.